### PR TITLE
Fix string type for buildargs API definition

### PR DESCRIPTION
**Pull request by sirreal**
**- What I did**

Update buildargs API swagger definition from `integer` to `string`

**- How I did it**

n/a

**- How to verify it**

n/a

**- Description for the changelog**

Docs - Correct `buildargs` API type to `string`

**- A picture of a cute animal (not mandatory but encouraged)**

Legit?

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -6174,7 +6174,7 @@ paths:
         - name: "buildargs"
           in: "query"
           description: "JSON map of string pairs for build-time variables. Users pass these values at build-time. Docker uses the buildargs as the environment context for commands run via the `Dockerfile` RUN instruction, or for variable expansion in other `Dockerfile` instructions. This is not meant for passing secret values. [Read more about the buildargs instruction.](https://docs.docker.com/engine/reference/builder/#arg)"
-          type: "integer"
+          type: "string"
```
Comment (thaJeztah) [inline api/swagger.yaml]: As a follow-up we could add an example value for this;

```json
{"ARG_ONE":"foo","ARG_TWO":"bar"}
```

or (URL encoded, but I don't think we need to show the encoded value):

```
%7B%22ARG_ONE%22%3A%22foo%22%2C%22ARG_TWO%22%3A%22bar%22%7D
```

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

File: api/swagger.yaml (hunk 1 above)
Comment (sirreal) [reply api/swagger.yaml]: Thanks. I'd be happy to add that here. I was a bit confused by the language so a better description would be 💯

Comment (thaJeztah) [reply api/swagger.yaml]: Yes, that'd be great!

Looks like (unfortunately), ReDoc doesn't support example values for query parameters (at least, they don't seem to show in the output; tried both `example:` and `x-example`), so the example value best is included in the description.

For readability, it's probably best to change the description text to use the "multi line" format (also a bit easier to work with);

```yaml
description: |
  JSON map of string pairs for build-time variables. Users pass these values at
  build-time. Docker uses the buildargs as the environment context for commands
  run via the `Dockerfile` RUN instruction, or for variable expansion in other
  `Dockerfile` instructions. This is not meant for passing secret values.
  [Read more about the buildargs instruction](https://docs.docker.com/engine/reference/builder/#arg)"
```

**Comment by sirreal** [comment]
I pushed a few changes as discussed with @thaJeztah (multi-line format, example).

This is the current result:

buildargs

File: api/swagger.yaml [hunk 2]
```diff
@@ -6173,8 +6173,19 @@ paths:
           type: "integer"
         - name: "buildargs"
           in: "query"
-          description: "JSON map of string pairs for build-time variables. Users pass these values at build-time. Docker uses the buildargs as the environment context for commands run via the `Dockerfile` RUN instruction, or for variable expansion in other `Dockerfile` instructions. This is not meant for passing secret values. [Read more about the buildargs instruction.](https://docs.docker.com/engine/reference/builder/#arg)"
-          type: "integer"
+          description: >
```
Comment (thaJeztah) [inline api/swagger.yaml]: nit: for other multi-line blocks we use `|` (preserve newlines/whitespace)

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

**Comment by SamMousa** [comment]
Should this be backported so older swagger files are also fixed?
As it stands now any generated API client for the older API versions (up to 1.36) will not be fixed.

**Comment by thaJeztah** [comment]
There's likely other changes that ideally should be backported to older versions. I'm actually thinking of pulling in the swagger documents for older API versions to master (e.g. `swagger-1.35.yaml`, `swagger-1.34.yaml`) to make it easier to patch multiple versions. If someone can open a tracking issue for that so that we can coordinate that effort (may need some discussion with the docs team as well)

**Pull request merged**